src/parser/parser.yy

```diff
@@ -3289,14 +3302,14 @@ create_space_sentence
         sentence->setComment($5);
         $$ = sentence;
     }
-    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON name_label {
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON zone_name_list {
```

Comment:
missing pattern that create space without zone_name_list? I see your code implementation that create space without zone_name_list means create space on all zones.